vision_datasets/key_value_pair/manifest.py

```diff
@@ -0,0 +1,140 @@
+
+
+class KeyValuePairSchema:
+    def __init__(self, name: str, field_schema_dict: Dict, description: str = None) -> None:
+        self.name = name
+        self.description = description
+        self.field_schema = {k: KeyValuePairFieldSchema(**v) for k, v in field_schema_dict.items()}
+    
+    def __eq__(self, other) -> bool:
+        return isinstance(other, KeyValuePairSchema) and self.name == other.name and self.field_schema == other.field_schema and self.description == other.description
+
+
+class KeyValuePairLabelManifest(MultiImageLabelManifest):
+    """
+    {
+        "key_value_pairs": {"key1": "value1", ...},
+        "text": "optional text input for this annotation"
+    }
+    """
+    LABEL_KEY = 'key_value_pairs'
+    INPUT_KEY = 'text'
+    
+    @property
+    def key_value_pairs(self) -> dict:
+        return self.label_data[self.LABEL_KEY]
+
+    @property
+    def text(self) -> Optional[dict]:
+        return self.label_data.get(self.INPUT_KEY, None)
+
+    def _read_label_data(self):
+        raise NotImplementedError('Read label data is not supported!')
+    
+    def _check_label(self, label_data):
+        if not isinstance(label_data, dict) or self.LABEL_KEY not in label_data:
+            raise ValueError
+
+    @classmethod
+    def check_schema_match(cls, key_value_pairs: Dict, schema: KeyValuePairSchema):
+        for key, field_schema in schema.field_schema.items():
+            if key not in key_value_pairs:
+                raise ValueError(f'{key} not found')
+            KeyValuePairLabelManifest.check_field_schema_match(key_value_pairs[key], field_schema)
+
+    @classmethod
+    def check_field_schema_match(cls, value, field_schema: KeyValuePairFieldSchema):
+        if not isinstance(value, KeyValuePairFieldSchema.TYPE_NAME_TO_PYTHON_TYPE[field_schema.type]):
+            raise ValueError(f'{value} is not of type {field_schema.type}')
+        if field_schema.enum:
+            if value not in field_schema.enum:
+                raise ValueError(f'{value} not in enum {field_schema.enum}')
+        if field_schema.type == 'boundingBox':
+            if len(value) != 4:
+                raise ValueError('boundingBox must have 4 elements')
+        elif field_schema.type == 'array':
+            for array_item in value:
+                cls.check_field_schema_match(array_item, field_schema.items)
+        elif field_schema.type == 'object':
+            for k, v in value.items():
+                if k not in field_schema.properties:
+                    raise ValueError(f'{k} not found in schema')
+                cls.check_field_schema_match(v, field_schema.properties[k])
+
+
+class KeyValuePairDatasetManifest(DatasetManifestWithMultiImageLabel):
+    """Manifest that has schema in additional_info which defines the structure of the key-value pairs in the annotations."""
+
+    def __init__(self, images, annotations, additional_info):
+        if 'schema' not in additional_info:
```

Comment:
Is it possible to take schema directly from the ctor? additional_info is usually stored with dataset-specific metadata,